### PR TITLE
feat: send performance timing values to GTM

### PR DESCRIPTION
**Pull request by emmenko**
See #1247 

image

TODOs:
- [ ] decide what to do with custom dimensions for tracking `applicationName` and `projectKey`
- [ ] configure tracking event `TimingEvent` in GTM

### CODE REVIEW

File: packages/application-shell/src/track-performance.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            category: 'Navigation Timing',
+            variable: metricName,
+            value: data.timeToFirstByte,
+            label: 'Time To First Byte',
+          });
+          trackTiming({
+            category: 'Navigation Timing',
+            variable: metricName,
+            value: data.dnsLookupTime,
+            label: 'DSN lookup time',
```
Comment (adnasa) [inline packages/application-shell/src/track-performance.js]: nit: this is `DNS`, right?

Comment (emmenko) [reply packages/application-shell/src/track-performance.js]: Yes thanks

**Comment by emmenko** [comment]
Will pick this up in the next days as I need to make some changes in the GTM settings first.

**Comment by emmenko** [comment]
image

**Comment by emmenko** [comment]
I also added the custom dimensions (only on staging for now):

image

**Comment by tdeekens** [comment]
> I also added the custom dimensions (only on staging for now):

What would happen if those dimensions don't exist on one environment? I think GCP Prod EU is mostly/only used to get tracking insight so far. So it might be fine to also for know just have it there?

**Comment by emmenko** [comment]
The dimensions are optional, in a sense that if they don't get mapped/configured nothing happens, even if we "push them to gtm".

**Comment by tdeekens** [comment]
Then we might only need to configure them on EU (maybe US) and from my view that should be enough. What do you think?

**Comment by emmenko** [comment]
Sure, although I would like to configure gtm to be same in all envs. It's not much to do, I can show you if you want

**Comment by tdeekens** [comment]
> Sure, although I would like to configure gtm to be same in all envs. It's not much to do, I can show you if you want

Yeah true. Just wanted to reduce the amount of work if nobody uses it in the end. Maybe we should add it to one of the confluence sites how to setup a new env so we also do it for new envs?

**Comment by emmenko** [comment]
Already documented 😉

**Comment by adnasa** [comment]
> It's not much to do, I can show you if you want

I'm in on this

**Comment by tdeekens** [comment]
> Already documented 😉

Yup, but we might want to add these from this PR there too, right?

**Comment by emmenko** [comment]
FYI: I updated the document.

**Comment by emmenko** [comment]
Alright, I updated the GA/GTM settings to accept the new events

**Comment by emmenko** [comment]
As a follow up, I'll deprecate the `MeasureFirstPaint` component, which we can remove in the #1248

**Pull request merged**